File: frontend/src/types/activity.ts
import type { User } from './user';

export type ActivityType =
  | 'issue_created'
  | 'status_changed'
  | 'assigned'
  | 'commented'
  | 'priority_changed';

export interface Activity {
  id: string;
  type: ActivityType;
  actor: User;
  targetId: string;
  targetKey: string;
  targetTitle: string;
  detail: string;
  createdAt: string;
  projectId: string;
}

File: backend/jeerai-backend/src/main/java/com/jeerai/backend/model/Activity.java
package com.jeerai.backend.model;

import java.time.Instant;

import lombok.AllArgsConstructor;
import lombok.Data;
import lombok.NoArgsConstructor;

@Data
@NoArgsConstructor
@AllArgsConstructor
public class Activity {
    private String id;
    private String type;
    private User actor;
    private String targetId;
    private String targetKey;
    private String targetTitle;
    private String detail;
    private Instant createdAt;
    private String projectId;
}
